cmake/setup_compiler_flags_gnu.cmake

```diff
@@ -22,20 +22,30 @@
 
 IF( CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND
     CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9" )
-  MESSAGE(WARNING "\n"
+  MESSAGE(FATAL_ERROR "\n"
     "deal.II requires support for features of C++14 that are not present in\n"
     "versions of GCC prior to 4.9."
```

Comment:
You can upgrade this to "5.0".

Reply:
Just checking: Did we come to this agreement during the hackathon? If so, then this part of the documentation is outdated and I'll add a commit that updates it as well.

Reply:
I went ahead and pushed a change for this, and amended what I had previously with some other documentation changes along with the addition of the CMake policy check. We have a similar check in place a little further down in `setup_compiler_flags_gnu.cmake`